### PR TITLE
Use GitHub Actions + General Updates

### PR DESCRIPTION
**Pull request by JacksonMaxfield**
This is a large set of changes to the Modeling Cookiecutter template

1. Uses GH Actions for builds, tests, lints, and publishing to PyPI
2. More documentation and imo easier to follow documentation on how to use this template
3. Simpler `Makefile`; `make build`, `make clean`, and `make docs` are the only commands
4. Sphinx now handles markdown which means `make docs` now just works properly
5. General cleanup

I would say this closes:
#2: You can simply delete the workflow files if you don't want builds
#8: No need for setup instructions as build is just integrated with repo, but I did add comments about the build system
#11: Markdown is now directly supported and README and CONTRIBUTING have both made markdown
#18: I got rid of as many files as I could. We could get rid of more but the ones I kept make sense to me

If you want to test it out:
```
git clone git@github.com:AllenCellModeling/cookiecutter-pypackage.git
cd cookiecutter-pypackage/
git checkout use-gh-actions
cd ..
pip install cookiecutter
cookiecutter cookiecutter-pypackage/
```

If you don't want to test it out, please review just the markdown files.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,23 +1,8 @@
...
-
-- [ ] Provide tests for your pull request changes.
-
-- [ ] Provides documentation for features added by your pull request.
+**Pull request recommendations:**
+- [ ] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
+- [ ] Link to any relevant issue in the PR description. Ex: _Resolves [gh-12], adds tiff file format support_
+- [ ] Provide context of changes.
+- [ ] Provide relevant tests for your feature or bug fix.
+- [ ] Provide or update documentation for any feature added by your pull request.
```
Comment (cdw) [inline .github/PULL_REQUEST_TEMPLATE.md]: 👍 So short

File: README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Cookiecutter PyPackage
+
+AICS Cookiecutter template for a Python package.
+
+## Features
+* Uses `tox` and `pytest` for local testing
+* Builds and tests on Windows, Mac, and Ubuntu on every branch and pull request commit using GitHub Actions
+* Releases your Python Package to PyPI when your push to `stable` using GitHub Actions
```
Comment (cdw) [inline README.md]: *you

File: README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    * _Include administrators_
+    * _Restrict who can push to matching branches_
+* Add the repo to your ReadTheDocs_ account + turn on the ReadTheDocs service hook.
+
+
+## Suggested Git Branch Strategy
+1. `master` is for the most up-to-date development, very rarely should you directly commit to this branch.
+2. `stable` is for releases only. When you want to release your project on PyPI, simple make a PR from `master` to
+`stable`, this template will handle the rest.
+3. Your day to day work should exist on branches separate from `master`. Even if it is just yourself working on the
```
Comment (cdw) [inline README.md]: Wait, do GitHub actions not run if you commit direct to master?

Comment (JacksonMaxfield) [reply README.md]: GitHub Actions run on `push` and on a CRON job on `master`, `stable` runs on every `push` to `stable`, all branches run on any `push` or `pull request` trigger. Regardless of direct commit or PR, those are the triggers. If you are curious, and I think I explained this poorly, reading the first 4 lines of the workflow definitions explains how the run.

Comment (JacksonMaxfield) [reply README.md]: Will add these comments to the README tho

File: cookiecutter.json [hunk 4]
```diff
@@ -7,9 +7,6 @@
   "project_short_description": "Python Boilerplate contains all the boilerplate you need to create a Python package.",
   "pypi_username": "{{ cookiecutter.github_username }}",
   "version": "0.1.0",
-  "use_pytest": "y",
-  "use_pypi_deployment_with_travis": "y",
-  "create_author_file": "y",
   "open_source_license": [
     "Allen Institute Software License",
     "MIT license",
```
Comment (heeler) [inline cookiecutter.json]: MIT license not BSD3? just currious

File: {{cookiecutter.project_slug}}/.github/workflows/publish.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel
+    - name: Build package
+      run: |
+        python setup.py sdist bdist_wheel
+    - name: Publish to PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: {{ cookiecutter.pypi_username }}
```
Comment (heeler) [inline {{cookiecutter.project_slug}}/.github/workflows/publish.yml]: I think (¯\_(ツ)_/¯) it's possible to combine these so that the common block doesn't have to be written each time and the additional action is just a job dependency. It could be more bother than it's worth though.

File: {{cookiecutter.project_slug}}/MANIFEST.in [hunk 6]
```diff
@@ -1,10 +1,6 @@
-{% if cookiecutter.create_author_file == 'y' -%}
-include AUTHORS.rst
-{% endif -%}
-include CONTRIBUTING.rst
-include HISTORY.rst
+include CONTRIBUTING.md
 include LICENSE
-include README.rst
+include README.md
```
Comment (heeler) [inline {{cookiecutter.project_slug}}/MANIFEST.in]: CODE_OF_CONDUCT?

File: README.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
+# Cookiecutter PyPackage
+
```
Comment (heeler) [inline README.md]: Badge for license

File: {{cookiecutter.project_slug}}/docs/conf.py [hunk 8]
```diff
@@ -170,13 +176,10 @@
...
-     u'{{ cookiecutter.project_name }} Documentation',
+    (master_doc, "{{ cookiecutter.project_slug }}",
+     u"{{ cookiecutter.project_name }} Documentation",
      author,
-     '{{ cookiecutter.project_slug }}',
-     'One line description of project.',
-     'Miscellaneous'),
+     "{{ cookiecutter.project_slug }}",
+     "One line description of project.",
+     "Miscellaneous"),
```
Comment (heeler) [inline {{cookiecutter.project_slug}}/docs/conf.py]: Why did you change to double quotes? Stylistic or other reason

File: README.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+AICS Cookiecutter template for a Python package.
+
+## Features
+* Uses `tox` and `pytest` for local testing
+* Builds and tests on Windows, Mac, and Ubuntu on every branch and pull request commit using GitHub Actions
```
Comment (heeler) [inline README.md]: Builds and triggers unit tests | test suite on Wind...

File: README.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
+# Cookiecutter PyPackage
+
+AICS Cookiecutter template for a Python package.
+
+## Features
+* Uses `tox` and `pytest` for local testing
```
Comment (heeler) [inline README.md]: local testing is configured to use tox and pytest to trigger testing simply run tox from the command prompt in the project home folder.

File: {{cookiecutter.project_slug}}/CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+helps, and credit will always be given.
+
+## Get Started!
+Ready to contribute? Here's how to set up `{{ cookiecutter.project_slug }}` for local development.
+
+* Fork the `{{ cookiecutter.project_slug }}` repo on GitHub.
+* Clone your fork locally:
+
+```
+$ git clone git@github.com:{your_name_here}/{{ cookiecutter.project_slug }}.git
```
Comment (heeler) [inline {{cookiecutter.project_slug}}/CONTRIBUTING.md]: just to be careful you might want to change this to 
$ git clone --recurse-submodules git@github.com:...

So that if there are submodules they get checked out too. (if there aren't it won't do any harm)

File: {{cookiecutter.project_slug}}/.github/workflows/publish.yml (hunk 5 above)
Comment (JacksonMaxfield) [reply {{cookiecutter.project_slug}}/.github/workflows/publish.yml]: Discussed in person. For now, I am okay with duplicate jobs as they are pretty minimal. Something to look at for future for sure though.

File: {{cookiecutter.project_slug}}/MANIFEST.in (hunk 6 above)
Comment (JacksonMaxfield) [reply {{cookiecutter.project_slug}}/MANIFEST.in]: Will add one!

File: README.md (hunk 7 above)
Comment (JacksonMaxfield) [reply README.md]: Actually, yea I am going to convert this template to MIT License and put a badge. This template should be AICS licensed...

File: {{cookiecutter.project_slug}}/docs/conf.py (hunk 8 above)
Comment (JacksonMaxfield) [reply {{cookiecutter.project_slug}}/docs/conf.py]: Just stylistic reasoning

File: {{cookiecutter.project_slug}}/CONTRIBUTING.md (hunk 11 above)
Comment (JacksonMaxfield) [reply {{cookiecutter.project_slug}}/CONTRIBUTING.md]: Smart

**Comment by donovanr** [comment]
looks good to me
